presto-hive/src/test/java/io/prestosql/plugin/hive/TestShowStats.java

```diff
@@ -191,6 +191,12 @@ public void testShowStatsWithMultipleFromFails()
         assertQueryFails("SHOW STATS FOR (SELECT * FROM nation_partitioned, region)", ".*There must be exactly one table in query passed to SHOW STATS SELECT clause");
     }
 
+    @Test
+    public void testShowStatsWithSubqueryFails()
+    {
+        assertQueryFails("SHOW STATS FOR (SELECT * FROM (VALUES 1))", ".*There must be exactly one table in query passed to SHOW STATS SELECT clause");
```

Comment:
add also a test case:

`assertQueryFails("SHOW STATS FOR (SELECT * FROM (SELECT * FROM nation))", `

btw i think the case you have currently doesn't match the test method name -- it's more like `testShowStatsWithoutTableScan`